dist2src/core.py

```diff
@@ -412,7 +447,44 @@ def rebase_patches(self, from_branch, to_branch):
             self.source_git.checkout(from_branch)
             self.source_git.repo.git.reset("--hard", "HEAD^")
 
-        self.source_git.repo.git.rebase("--root", "--onto", to_branch)
         self.source_git.checkout(to_branch)
-        self.source_git.repo.git.merge("--ff-only", "-q", from_branch)
-        self.source_git.repo.git.branch("-d", from_branch)
+        commits_to_cherry_pick = [
+            c.hexsha[:8]  # shorter format for better readability in case of an error
+            for c in self.source_git.repo.iter_commits(from_branch)
+        ][-2::-1]
+        if commits_to_cherry_pick:
+            self.source_git.repo.git.cherry_pick(
+                *commits_to_cherry_pick,
+                keep_redundant_commits=True,
+                allow_empty=True,
+                strategy_option="theirs",
+            )
+        self.source_git.repo.git.branch("-D", from_branch)
+
+    def update_source_git(self, origin_branch: str, dest_branch: str):
```

Comment:
Thinking out loud (or, thinking out so you can see me thinking :D):

With the current design, in the update automation service we'd use the `update` command while the onboarding script would use `convert` - do we want that? I was thinking about a more convenient solution to only have a single command which cover both:
```python
if source_git_repo.exists:
  update()
else:
  convert()
```

Reply:
Yes, that's what I was asking last week on arch-meetings..;)

- There wasn't a strong opinion.
- The main concern was the conversion script which is API class, not CLI.


I'll probably pick one CLI subcommand but the condition would be about existance of the branch, WDYT?

Reply:
Done, PTAL.

Reply:
yes to the existance of the branch